postCheckReport.js

```diff
@@ -0,0 +1,252 @@
+  const channelUsers = await getChannelUsers('C5CU14MRS');
+  const messages = await getMessages('C5CU14MRS');
+  const messageUsers = messages.map((message) => message.user);
+  const absentUsers = channelUsers.filter(
+    (user) => !messageUsers.includes(user)
+  );
+  const absentUsersWithDetail = absentUsers.map((userId) => ({
+    id: userId,
+    realName: allUsersDic[userId].profile.real_name,
+  }));
+  console.log("original absend user list", absentUsersWithDetail)
+  const humiEmployees = await getHumiEmployeesList();
+
+
+  const timeOffUsersList = await getHumiEmployeeTimeOffList(startDate,endDate);
+
+  // Functions to get the employee and timeOffUsers data
+  // Should be applied to the actual API data
+  console.log('humi employees', humiEmployees.length);
+  const employees = humiEmployees.map((element) => {
+    const employee = element['attributes'];
+
+    const { id, first_name, last_name, end_date } = employee;
+     if(end_date === null)
+    return { id, first_name, last_name };
+     else return null;
+  }).filter((element) => element !== null);
+
+  console.log('humi employees', employees.length);
+  ;
+  const timeOffUsers = timeOffUsersList.map((element) => {
+    const employee = element['attributes'];
+    const { employee_id, status, start_at, end_at } = employee;
+    return { employee_id, status, start_at, end_at };
+  });
+
+  // timeOffUsers don't have names associated with them, so we need to merge the two arrays on ID
+  // Pretty messy but works, maybe can refactor later
+
+  const timeOffUsersWithNames = employees.filter(({ id: id1 }) => timeOffUsers.some(({ employee_id: id2 }) => id2 === id1))
+  .map((element) => {
+    const { first_name, last_name } = element;
+    return {
+      realName: `${first_name} ${last_name}`,
+      ...element,
+    };
+  });
+
+
+  console.log('timeOffUsersWithNames',timeOffUsersWithNames)
+
+  // Filter the slack absent users
+
+  const humiNamesOnly = timeOffUsersWithNames.map(
+    (element) => element.realName
+  );
+  const absentUsersList = absentUsersWithDetail.filter((element) =>
+    !humiNamesOnly.includes(element.realName)
+  );
+  // Here are C-Level manager, they do not need to check-in everyday
+ const alwaysNotSendNotificationUsers = [
+      'Sam',
+      'Rohit Boolchandani',
+      'Shaun Jamieson',
+      'freshtimescounter',
+
+ ]
+ const finalAbsentUsersList = absentUsersList.filter((element) =>
+    !alwaysNotSendNotificationUsers.includes(element.realName )
+  );
+  // call helper func to eliminate all the user on vacation or on leave at here
+  console.log('test finalAbsentUsersList', finalAbsentUsersList);
+// in order to check if today is a holiday, we do a simple trick here,
+// if more than half employees does not show up, we suppose this is a holiday
+
+if(employees.length <=  finalAbsentUsersList.length * 2 ) {
+  console.log('Today is a holiday! no one is on');
+} else {
+
+    const text = `These user ${finalAbsentUsersList.map(
```

Comment:
As far as I understand, this file differs mostly by this function, right? 
Maybe we could pull all the code from here and hackathon.js into a separate file and either make a class or just export select-few functions; then create two files for cronjobs, import that class/file, call init that does all the similar initialization stuff, and then call just that one function that differs? 

Not a pro at this, but that's the idea that came to mind.